packages/api/src/car.js

```diff
@@ -273,6 +273,13 @@ async function carStat (carBlob) {
     if (blockSize > MAX_BLOCK_SIZE) {
       throw new InvalidCarError(`block too big: ${blockSize} > ${MAX_BLOCK_SIZE}`)
     }
+    if (block.cid.multihash.code !== sha256.code) {
```

Comment:
One of the cases where is not good to support everything. While it is the default, we support other hash functions, including by `ipfs-car` itself: https://github.com/web3-storage/ipfs-car/blob/main/src/pack/index.ts#L21 

Considering we recommend `ipfs-car` and people can pack their own CARs, as well as we don't inform anywhere to the users that we only support in web3.storage sha256 we need to be careful here.

What do you think @olizilla ? Should we support at least the hash functions that are in multiformats repo and properly document it? Or just sha256?

Reply:
From discussion with @alanshaw and @vasco-santos we're going to

- adjust this PR so that we don't error if we see a non-sha256 hash
- land this PR so that sha256 hashed blocks (the vast majority) *are* validated
- open an issue to see what other hashes we want to pre-validate _(ideally many, but bundle size and perf of hashing functions other than sha2 need to be considered given that the impl is built in to subtle.crypto and may be hardware optimised)_